src/cryptography/hazmat/primitives/kdf/kbkdf.py

```diff
@@ -0,0 +1,160 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+from __future__ import absolute_import, division, print_function
+
+import binascii
+import struct
+
+import six
+
+from cryptography import utils
+from cryptography.exceptions import (
+    AlreadyFinalized, InvalidKey, UnsupportedAlgorithm, _Reasons
+)
+from cryptography.hazmat.backends.interfaces import HMACBackend
+from cryptography.hazmat.primitives import constant_time, hashes, hmac
+from cryptography.hazmat.primitives.kdf import KeyDerivationFunction
+
+
+@utils.register_interface(KeyDerivationFunction)
+class KBKDFHMAC(object):
+    COUNTER_MODE = 'ctr'
+
+    LOCATION_BEFORE_FIXED = 'before_fixed'
+    LOCATION_AFTER_FIXED = 'after_fixed'
+
+    def __init__(self, algorithm, mode, length, rlen, llen,
+                 location, label, context, fixed, backend):
+        if not isinstance(backend, HMACBackend):
+            raise UnsupportedAlgorithm(
+                "Backend object does not implement HMACBackend.",
+                _Reasons.BACKEND_MISSING_INTERFACE
+            )
+
+        if not isinstance(algorithm, hashes.HashAlgorithm):
+            raise UnsupportedAlgorithm(
+                "Algorithm supplied is not a supported hash algorithm.",
+                _Reasons.UNSUPPORTED_HASH
+            )
+
+        if (label or context) and fixed:
+            raise ValueError("When supplying fixed data, "
+                             "label and context is ignored.")
+
+        if mode is None:
```

Comment:
I'd prefer to require explicit passing of the counter mode arg and just check to make sure that the passed value is a member of the enum via `isinstance`.